File: dotnet-desktop-guide/winforms/input-mouse/snippets/how-to-distinguish-between-clicks-and-double-clicks/vb/Form1.vb
Imports System.Windows.Forms
Imports System.Drawing

'<RollbackForm>
Partial Public Class Form1

    Private _initialStyle As FormBorderStyle
    Private _isDoubleClicking As Boolean

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim button1 As New DoubleClickButton

        _initialStyle = FormBorderStyle

        button1.Location = New Point(50, 50)
        button1.Size = New Size(200, 23)
        button1.Text = "Click or Double Click"

        AddHandler button1.Click, AddressOf Button1_Click
        AddHandler button1.DoubleClick, AddressOf Button1_DoubleClick

        Controls.Add(button1)

    End Sub

    Private Sub Button1_DoubleClick(sender As Object, e As EventArgs)
        ' This flag prevents the click handler logic from running
        ' A double click raises the click event twice.
        _isDoubleClicking = True
        FormBorderStyle = _initialStyle
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs)
        If _isDoubleClicking Then
            _isDoubleClicking = False
        Else
            FormBorderStyle = FormBorderStyle.FixedToolWindow
        End If
    End Sub
End Class
'</RollbackForm>

File: dotnet-desktop-guide/winforms/input-mouse/snippets/how-to-distinguish-between-clicks-and-double-clicks/csharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Runtime.InteropServices;

namespace project
{
    //<RollbackForm>
    public partial class Form1 : Form
    {
        private FormBorderStyle _initialStyle;
        private bool _isDoubleClicking;

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            _initialStyle = this.FormBorderStyle;

            var button1 = new DoubleClickButton();
            button1.Location = new Point(50, 50);
            button1.Size = new Size(200, 23);
            button1.Text = "Click or Double Click";
            button1.Click += Button1_Click;
            button1.DoubleClick += Button1_DoubleClick;

            Controls.Add(button1);
        }

        private void Button1_DoubleClick(object sender, EventArgs e)
        {
            // This flag prevents the click handler logic from running
            // A double click raises the click event twice.
            _isDoubleClicking = true;
            FormBorderStyle = _initialStyle;
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (_isDoubleClicking)
                _isDoubleClicking = false;
            else
                FormBorderStyle = FormBorderStyle.FixedToolWindow;
        }
    }
    //</RollbackForm>
}
